pkg/kubectl/cmd/util/helpers.go

```diff
@@ -142,6 +143,9 @@ func checkErr(err error, handleErr func(string, int)) {
 		}
 	case clientcmd.IsConfigurationInvalid(err):
 		handleErr(MultilineError("Error in configuration: ", err), DefaultErrorExitCode)
+	case discovery.IsParseServerVersionFailedError(err):
```

Comment:
why is the `default:` error handling below not enough?

Reply:
`default:` error handling below uses the type of error for comparison.

Reply:
default error handling which one you mean? there are two default there.

Reply:
We're already matching the error's type in the nested switch below. You could just add a new `case *ErrParseServerVersionFailed:`. Also, if you're already wrapping the error in `&ErrParseServerVersionFailed{}`, and its `Error` method already prepends the message `unable to parse the server version`, I don't think it's necessary to state that again like you're doing in the line below.

Reply:
@juanvallejo do you mean by returning the error message `unable to parse the server version` is that enough?

Reply:
updated